tfrog-motordriver/main.c

```diff
@@ -655,7 +655,8 @@ int main()
   // Driver loop
   while (1)
   {
-    AT91C_BASE_WDTC->WDTC_WDCR = 1 | 0xA5000000;
+    if (!HEARTBEAT)
+      AT91C_BASE_WDTC->WDTC_WDCR = 1 | 0xA5000000;
```

Comment:
The watchdog timer should be reset automatically before entering `servo_level > SERVO_LEVEL_STOP`.
Otherwise, the device periodically resets its-self including USB connection which is not good behavior.

Reply:
it's continuously checked in timer0_vel_calc(), right?
so, you mean, it's better to switch servo_level to STOP if heartbeat doesn't arrive?

Reply:
After the power on, `servo_level` goes `SERVO_LEVEL_STOP` until PC (yp-spur) starts communicating with the device and changes control mode to `SERVO_LEVEL_TORQUE` or `SERVO_LEVEL_VELOCITY`.

If watchdog time is not cleared automatically on `SERVO_LEVEL_STOP`, the device triggers reset for every watchdog duration.
This resets USB connection and make PC difficult to start the communication.

> so, you mean, it's better to switch servo_level to STOP if heartbeat doesn't arrive?

That's not what I would like to say in the previous comment.
But It is sounds also good to make the device always under control. (Everything is uncontrollable during reset.)

Reply:
As we discussed, I'm gonna stop the motor from the power cycle via a I/O pin. According to the hardware team, a corresponded pin has not been decided yet.